src/EditorFeatures/Core/Shared/Preview/PreviewSolutionCrawlerRegistrationService.cs

```diff
@@ -17,14 +18,17 @@ namespace Microsoft.CodeAnalysis.Editor.Shared.Preview
     internal class PreviewSolutionCrawlerRegistrationServiceFactory : IWorkspaceServiceFactory
     {
         private readonly DiagnosticAnalyzerService _analyzerService;
+        private readonly IAsynchronousOperationListener _listener;
 
         [ImportingConstructor]
-        public PreviewSolutionCrawlerRegistrationServiceFactory(IDiagnosticAnalyzerService analyzerService)
+        public PreviewSolutionCrawlerRegistrationServiceFactory(IDiagnosticAnalyzerService analyzerService, IAsynchronousOperationListenerProvider listenerProvider)
```

Comment:
this is workspace service specific to Preview Kind. the test doesn't create PreviewKind workspace. if you want this, you should create PreviewKind workspace.

Reply:
➡️ It's already created in `PreviewFactoryService.CreateChangedDocumentPreviewViewAsync`.

Reply:
ah, I see good then.